backend/matching-service/controllers/matching.controller.js

```diff
@@ -3,6 +3,9 @@ const mQueue = new queue();
 const matchObject = require('../models/match.object.js');
 
 exports.enqueue = (req, res) => {
+    if (!req.session) {
```

Comment:
Yup you're right, I thought of this method first to check at the controller but I decided to go with the other method to force all tokens to be verified first rather than just trusting the cookie. This was me forgetting to remove this older implementation, my bad.